cmds/modules/zui/service.go

```diff
@@ -0,0 +1,149 @@
+package zui
+
+import (
+	"context"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+	"github.com/threefoldtech/zbus"
+
+	"github.com/threefoldtech/zos/pkg/stubs"
+)
+
+const (
+	activeStatus     = "Active"
+	InProgressStatus = "In progress"
+	FailedStatus     = "Failed"
+	InactiveStatus   = "Inactive"
+)
+
+func serviceRender(ctx context.Context, client zbus.Client, grid *ui.Grid, render *signalFlag) (chan bool, error) {
+	services := widgets.NewTable()
+
+	grid.Set(
+		ui.NewRow(1,
+			ui.NewCol(1, services),
+		),
+	)
+
+	services.Title = "Services Status"
+	services.RowSeparator = false
+
+	services.Rows = [][]string{
+		{"", "Status"},
+		{"Networkd", InProgressStatus},
+		{"Registerar", InactiveStatus},
+		{"Statistics", InactiveStatus},
+		{"Containerd", InactiveStatus},
+		{"Storaged", InactiveStatus},
+		{"Noded", InactiveStatus},
+		{"Powerd", InactiveStatus},
+	}
+
+	render.Signal()
+
+	done := make(chan bool)
+
+	go func() {
+		networkStatus := make(chan string)
```

Comment:
isn't it better if we have only one chan that all services write to it and we check the returned string against which service is up or so?

Reply:
I agree